### DISCUSSION TITLE
styling display To-Do task

### DISCUSSION THREAD

**Discussion by plegrand1** [gsantner/markor#1639] [Q&A] [answered]
Hello, 
Is there à way to modify the way Markor display task in To-Do on display mode? 
Css or something else? 
Thanks for your help

**Comment by plegrand1** [comment, marked as answer]
I see that it is possible to modify the stylesheet in the View Mode with "Inject -> head".
Does this allow to change the display mode of Todo?
How do you use it? Is there a help page for this feature?
Thanks for your help

**Reply by gsantner** [reply]
Yes inject allows to modify & style any view mode.

How it works? It is CSS, so basically you need to learn a bit of programming. <https://www.w3schools.com/css/>

Altought for todo.txt the main focus is on the edit mode. The view mode is  more or less dumb one line of todo, one line of text, without sugar coat.

**Reply by alexandre-mbm** [reply]
Does work for PDF export? I want to get a PDF with no margin.

**Reply by gsantner** [reply]
PDF export is more or less 99% what you see in the View-Mode. So if it works there it should look quite the same on the PDF/print export.

**Comment by plegrand1** [comment]
Hello and thank you for your answer,
in case I want to change the look of the toto in view mode, what are the tags to use and does the css file have to have a specific name.
It's not really the CSS that I have a problem with but more the way to set it up in markor.
In inject -> head I add this line : 
`<link rel="stylesheet" href="style.css" type="text/css" />` ?
But then what are the specific tags that match the todo?

Thanks again
Pascal

**Reply by Russe** [reply]
To see which tags are used in view mode, you can export a todo file with "Share > HTML (as plain text)".

**Reply by amelillo** [reply]
Hello ^_^
I tried to inject code to change the **asciidoc** default rendering (dark mode) using this snippet:
`<style>
body {
           background-color: white;
           }
</style>`
but it seemed not to work.
Then I discovered that changing "white" to "black" actually had an effect in view mode but just for the **Markdown** files!

How can I tell Markor to override defaults in the **asciidoc** documents and **not** in the markdown ones?

and, by the way, wouldn't it be easier to let Markor choose a css stylesheet instead of transcribing lines and lines of css? Something like "override default stylesheet" which lets the user load his own css file and that is maybe used also elsewhere (in a website, web app, desktop editor or whatever).

**Reply by Russe** [reply]
When you export an asciidoc with "Share > HTML (as plain text)", then you can see that body has class "format-asciidoc fileext-adoc".
Maybe you can use these classes to use css for asciidoc files only.

**Reply by amelillo** [reply]
Thank you. Just didn't notice that because it's at the end of the file.
I added the class reference directly taken from an export, and the result was that when switching to View mode it just flashes white for a moment and then reverts to the default dark css mode. Here's what I first tried:
`<style> body { class: "format-asciidoc fileext-adoc"; background-color: white; } </style>`
Then I used:
`<style> body.format-asciidoc { background-color: white; } </style>`
And it finally worked, but how can CSS attribute a class to an element if it's just supposed to style it? Shouldn't I do that with a regular HTML class assignment?
And, again, wouldn't it be simpler to let Markor just apply a ready-made css stylesheet instead of messing around with a reformulation of the same statements? If the default css was accessible (i.e. with a non rooted device) I'd just tweak it instantly backing up the file and modifying the default. It would also be much more readable as every information would be in one place.

**Reply by gsantner** [reply]
@amelillo There were some asciidoc related theme changes merged. You can try a development version from here <https://github.com/gsantner/markor/actions/runs/6152028911>. Maybe it helps somehow also as I see you want to make things white for asciidoc.

**Comment by amelillo** [comment]
> Maybe it helps somehow also as I see you want to make things white for asciidoc.

And I did it, as a start. My asciidoc notes are now rendered plain white. Now I just have to tune it to my needs, that's adapting an already working css I use to style my asciidoc files anywhere else.
It's not a major concern as for important files I can apply my css externally and in Markor I can safely ignore the details. But it's additional work, though it's also part of the fun ^_^
